### PR TITLE
SMS package

### PR DESCRIPTION
**Pull request by kfedor**

### CODE REVIEW

File: sms/contract.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// Public API.
+
+type (
+	Parcel struct {
+		ToNumber string
+		Message  string
+	}
+
+	Client interface {
+		Send(ctx context.Context, parcel Parcel) error
```
Comment (victor-oancea-leftclick-io) [inline sms/contract.go:20]: pls always define methods in interfaces without arguments

File: sms/sms.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+	for _, a := range parcels {
+		copyA := a
+		g.Go(func() error {
+			err := s.Send(ctx, copyA)
+			if err != nil {
+				return err
+			}
+
+			return nil
```
Comment (victor-oancea-leftclick-io) [inline sms/sms.go:59]: return s.Send

File: sms/sms.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return errors.Wrapf(err, "error during sending multiple messages")
+	}
+
+	return nil
```
Comment (victor-oancea-leftclick-io) [inline sms/sms.go:67]: return errors.wrapf(g.wait

**Review by victor-oancea-leftclick-io — CHANGES_REQUESTED** [review]

**Review by nusanovkornilov-leftclick — CHANGES_REQUESTED** [review]

File: sms/sms.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+func (s *sms) Send(ctx context.Context, parcel Parcel) error {
+	return errors.Wrapf(retry(ctx, func() error {
+		if ctx.Err() != nil {
+			//nolint:wrapcheck // It's a proxy.
+			return backoff.Permanent(ctx.Err())
+		}
+
+		_, err := s.client.Api.CreateMessage(new(openapi.CreateMessageParams).
```
[the remark this answers is not in the document]
Comment (kfedor) [reply sms/sms.go:38]: response info is 
type ApiV2010Message struct {
	// The SID of the Account that created the resource
	AccountSid *string `json:"account_sid,omitempty"`
	// The API version used to process the message
	ApiVersion *string `json:"api_version,omitempty"`
	// The message text
	Body *string `json:"body,omitempty"`
	// The RFC 2822 date and time in GMT that the resource was created
	DateCreated *string `json:"date_created,omitempty"`
	// The RFC 2822 date and time in GMT when the message was sent
	DateSent *string `json:"date_sent,omitempty"`
	// The RFC 2822 date and time in GMT that the resource was last updated
	DateUpdated *string `json:"date_updated,omitempty"`
	// The direction of the message
	Direction *string `json:"direction,omitempty"`
	// The error code associated with the message
	ErrorCode *int `json:"error_code,omitempty"`
	// The description of the error_code
	ErrorMessage *string `json:"error_message,omitempty"`
	// The phone number that initiated the message
	From *string `json:"from,omitempty"`
	// The SID of the Messaging Service used with the message.
	MessagingServiceSid *string `json:"messaging_service_sid,omitempty"`
	// The number of media files associated with the message
	NumMedia *string `json:"num_media,omitempty"`
	// The number of messages used to deliver the message body
	NumSegments *string `json:"num_segments,omitempty"`
	// The amount billed for the message
	Price *string `json:"price,omitempty"`
	// The currency in which price is measured
	PriceUnit *string `json:"price_unit,omitempty"`
	// The unique string that identifies the resource
	Sid *string `json:"sid,omitempty"`
	// The status of the message
	Status *string `json:"status,omitempty"`
	// A list of related resources identified by their relative URIs
	SubresourceUris *map[string]interface{} `json:"subresource_uris,omitempty"`
	// The phone number that received the message
	To *string `json:"to,omitempty"`
	// The URI of the resource, relative to `https://api.twilio.com`
	Uri *string `json:"uri,omitempty"`
}

we check error and its enough :))

**Review by victor-oancea-leftclick-io — DISMISSED** [review]

File: sms/sms.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	g, ctx := errgroup.WithContext(ctx)
+
+	for _, a := range parcels {
+		copyA := a
+		g.Go(func() error {
+			return s.Send(ctx, copyA)
+		})
+	}
+
+	return errors.Wrapf(g.Wait(), "error during sending multiple messages")
```
[the remark this answers is not in the document]
Comment (victor-oancea-leftclick-io) [reply sms/sms.go:58]: Nikolay is right, its a best effort, not a fail fast

[the remark this answers is not in the document]
Comment (kfedor) [reply sms/sms.go:58]: ok you want to collect all errors and return array of errors?

[the remark this answers is not in the document]
Comment (victor-oancea-leftclick-io) [reply sms/sms.go:58]: Nope, multierror em

**Review by victor-oancea-leftclick-io — APPROVED** [review]

**Review by nusanovkornilov-leftclick — DISMISSED** [review]

**Review by antonh21 — DISMISSED** [review]

**Review by antonh21 — DISMISSED** [review]

**Review by nusanovkornilov-leftclick — DISMISSED** [review]

**Comment by victor-oancea-leftclick-io** [comment]
@kfedor fix conflicts

**Pull request merged**